src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -126,6 +147,28 @@ public override void Initialize (AnalysisContext context)
 						ReportRequiresDiagnostic (operationContext, member, requiresAttribute);
 					}
 				}
+
+				void CheckOverridesAndInterfaces (
+					SymbolAnalysisContext symbolAnalysisContext,
+					ISymbol member)
+				{
+					ISymbol? overriddenMember;
+					ImmutableArray<ISymbol> interfacesImplementations;
+					if ((member.IsVirtual || member.IsOverride) && member.TryGetOverriddenMember (out overriddenMember)) {
+						if (!member.HasAttribute (RequiresAttributeName) && overriddenMember!.HasAttribute (RequiresAttributeName))
```

Comment:
I think you could fold these four if-else statements by using xor and deciding in `ReportMatchOverrideOrInterfaceDiagnostic` which of the two symbols is to blame (it will always be the one without the attribute).

```suggestion
						if (member.HasAttribute (RequiresAttributeName) ^ overriddenMember!.HasAttribute (RequiresAttributeName))
```